owncloudApp/src/main/java/com/owncloud/android/ui/activity/LogHistoryActivity.kt

```diff
@@ -0,0 +1,84 @@
+ *
+ *   @author David González Verdugo
+ *   Copyright (C) 2019 ownCloud GmbH.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License version 2,
+ *   as published by the Free Software Foundation.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.owncloud.android.ui.activity
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.owncloud.android.R
+import info.hannes.logcat.BothLogsFragment
+import info.hannes.logcat.LogcatFragment
+import info.hannes.timber.fileLoggingTree
+import timber.log.Timber
+
+class LogHistoryActivity : ToolbarActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContentView(R.layout.logs)
+        setupToolbar()
+
+        // Check that the activity is using the layout version with the fragment_container FrameLayout
+        if (findViewById<View>(R.id.fragment_container) != null) {
+
+            // However, if we're being restored from a previous state,
+            // then we don't need to do anything and should return or else
+            // we could end up with overlapping fragments.
+            if (savedInstanceState != null) {
+                return
+            }
+
+            var logFragment: Fragment? = null
+
+            Timber.forest().fileLoggingTree()?.let {
+                logFragment = BothLogsFragment.newInstance(
+                    bothLogsTargetFileName,
+                    bothLogsSearchHint,
+                    logCatSearchHint,
+                    getString(R.string.mail_logger)
+                )
+            } ?: run {
+                logFragment = LogcatFragment.newInstance(
+                    logCatTargetFileName,
+                    logCatSearchHint,
+                    getString(R.string.mail_logger)
+                )
+            }
+
+            // In case this activity was started with special instructions from an
+            // Intent, pass the Intent's extras to the fragment as arguments
+            // firstFragment.arguments = intent.extras
+
+            logFragment?.let {
+                supportFragmentManager.beginTransaction()
+                    .add(R.id.fragment_container, it).commit()
+            }
+        }
+    }
+
+    companion object {
+        private const val logCatTargetFileName = "logfile.log"
+        private const val logCatSearchHint = "search logcat"
+        private const val bothLogsTargetFileName = "Owncloud.log"
```

Comment:
Please, don't use `Owncloud` as hardcoded string inside the code, due to branding reasons. You can use the app_name or other parameter.

Reply:
ok, let's call it log.log (or something other less-meaningful) before it was
<img width="503" alt="image" src="https://user-images.githubusercontent.com/3314607/69242970-5365a800-0ba2-11ea-8fca-43661f46a047.png">
even _current_ and _older_ has that much meaningful name
<img width="412" alt="image" src="https://user-images.githubusercontent.com/3314607/69242987-5b254c80-0ba2-11ea-8ef4-7338aa0d45f1.png">

`app_name` needs context , so `companion object` is not the best place anymore then

But the branch in in your repo, you can do what you want